pkg/cloudprovider/aws/fleet/packing/nodecapacity.go

```diff
@@ -80,22 +94,20 @@ type nodeCapacity struct {
 	total        v1.ResourceList
 }
 
-func (nc *nodeCapacity) reserve(podSpec *v1.PodSpec) bool {
-	resources := scheduling.GetResources(podSpec)
-	cpu := nc.reserved.Cpu()
-	cpu.Add(*resources.Cpu())
-	memory := nc.reserved.Memory()
-	memory.Add(*resources.Memory())
-	podCount := nc.reserved.Pods()
-	podCount.Add(*resource.NewQuantity(1, resource.BinarySI))
+func (nc *nodeCapacity) reserve(resources v1.ResourceList) bool {
+	targetUtilization := binpacking.MergeResources(nc.reserved, resources)
 	// If pod fits reserve the capacity
-	if nc.total.Cpu().Cmp(*cpu) >= 0 &&
-		nc.total.Memory().Cmp(*memory) >= 0 &&
-		nc.total.Pods().Cmp(*podCount) >= 0 {
-		nc.reserved[v1.ResourceCPU] = *cpu
-		nc.reserved[v1.ResourceMemory] = *memory
-		nc.reserved[v1.ResourcePods] = *podCount
+	if nc.total.Cpu().Cmp(*targetUtilization.Cpu()) >= 0 &&
+		nc.total.Memory().Cmp(*targetUtilization.Memory()) >= 0 &&
+		nc.total.Pods().Cmp(*targetUtilization.Pods()) >= 0 {
+		nc.reserved = targetUtilization
 		return true
 	}
 	return false
 }
+
+func (nc *nodeCapacity) reserveForPod(podSpec *v1.PodSpec) bool {
```

Comment:
IMO, it would be cleaner if we always modeled these as pods
```
reserve(daemonpod)
reserve(workloadpod)
reserve(kubelet_overhead_modeled_as_pod)
```
It's a bit of a hack, but WDYT?

Minimally, if we're keeping both, I'd call it

reserve(pod)
reserveOverhead(resources)

since pod is the common case.

Reply:
```
reserve(daemonpod)
reserve(workloadpod)
reserve(kubelet_overhead_modeled_as_pod)
```

The problem with this is, workload pod needs to increment pod quantity in resources, but daemonset pods and kubelet don't need to specify this. Either the caller of this function will need to do this or we will need a wrapper around reserve to do this.

For the second suggestion, reserve(pod) will need to call reserveOverhead(resources)? Are we fine with this?